tests/Integration/RemoteAPI/RelatedRemoteAPITest.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * Parsely `/related` Remote API Integration tests.
+ *
+ * @package Parsely\Tests
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Tests\Integration;
+
+use Parsely\Parsely;
+use Parsely\RemoteAPI\Related_Caching_Decorator;
+use Parsely\RemoteAPI\Related_Proxy;
+use WP_Error;
+use WP_Object_Cache;
+
+/**
+ * Parsely `/related` Remote API tests.
+ */
+final class RelatedRemoteAPITest extends TestCase {
+	/**
+	 * Internal Parsely variable
+	 *
+	 * @var Parsely $parsely Holds the Parsely object
+	 */
+	private static $parsely;
+
+	/**
+	 * The setUp run before each test
+	 */
+	public function set_up(): void {
+		parent::set_up();
+
+		self::$parsely = new Parsely();
+	}
+
+	/**
+	 * Data provider for test_related_api_url().
+	 *
+	 * @return array[]
+	 */
+	public function data_related_api_url(): array {
+		return array(
```

Comment:
Consider using `yield` here instead of a giant array. See https://github.com/Parsely/wp-parsely/pull/383 which made the exact same switch.